### PR TITLE
Support UInt8Array and friends

### PR DESCRIPTION
**Pull request by swankjesse**
We don't yet support typed arrays of size other than byte.

### CODE REVIEW

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        + "array[2] = 53;"
+        + "array[3] = 0;"
+        + "array[4] = 9;"
+        + "array;");
+    assertThat(byteArray).asList()
+        .containsExactly((byte) 86, (byte) 7, (byte) 53, (byte) 0, (byte) 9);
+  }
+
+  @Test public void int16ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
```
Comment (JakeWharton) [inline quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: Shouldn't these be `short[]`?

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  @Test public void int32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
+        + "var array = new Int32Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void uint32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
```
Comment (JakeWharton) [inline quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: `int[]`...

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        + "var array = new Uint32Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void float32ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
```
Comment (JakeWharton) [inline quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: `float[]`...

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        + "var array = new Float32Array(1);"
+        + "array[0] = 867.5309;"
+        + "array;");
+    assertThat(byteArray).isNull();
+  }
+
+  @Test public void float64ArrayUnsupported() {
+    byte[] byteArray = (byte[]) quickJs.evaluate(""
```
Comment (JakeWharton) [inline quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: `double[]`?

Comment (JakeWharton) [reply quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: Or maybe just leave as `Object` for now?

**Review by JakeWharton — APPROVED** [review]

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java (hunk 1 above)
Comment (swankjesse) [reply quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: Yep.

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java (hunk 4 above)
Comment (swankjesse) [reply quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java]: Lemme use the actual type that makes sense.

File: quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  @Test public void int16ArrayUnsupported() {
+    short[] shortArray = (short[]) quickJs.evaluate(""
+        + "var array = new Int16Array(1);"
+        + "array[0] = 8675;"
+        + "array;");
+    assertThat(shortArray).isNull();
+  }
+
+  @Test public void uint16ArrayUnsupported() {
+    short[] shortArray = (short[]) quickJs.evaluate(""
```
Comment (swankjesse) [inline quickjs/common/test/app/cash/quickjs/JsTypedArraysTest.java:99]: This is the most interesting one in the bunch. Could be either `short` or `char`, depending on whether you think char is Java’s uint16 type or if it’s only useful for UTF-16.

**Pull request merged**